src/coreclr/pal/inc/rt/palrt.h

```diff
@@ -156,21 +156,23 @@ inline void *__cdecl operator new(size_t, void *_P)
 
 #define _WINNT_
 
-// C++ standard, 18.1.5 - offsetof requires a POD (plain old data) struct or
-// union. Since offsetof is a macro, gcc doesn't actually check for improper
-// use of offsetof, it keys off of the -> from NULL (which is also invalid for
-// non-POD types by 18.1.5)
 //
-// As we have numerous examples of this behavior in our codebase,
-// making an offsetof which doesn't use 0.
-
+// The draft C++ standard (as of Aug 2022) 17.2.4 says the language
+// support for offsetof() on anything other than a standard-layout class
+// is conditionally supported.
+// This is more generous than the C++ standards when this code was originally written 
+//
 // PAL_safe_offsetof is a version of offsetof that protects against an
-// overridden operator&
+// overridden operator&.  Note, however, C++-98 and newer
+// requires offsetof to work correctly even if operator& is overloaded.
+//
 
 #define FIELD_OFFSET(type, field) __builtin_offsetof(type, field)
```

Comment:
Same for `FIELD_OFFSET`. You can delete the definition and replace the few uses with `offsetof`.